ports/stm32/machine_i2s.c

```diff
@@ -788,10 +790,19 @@ STATIC void machine_i2s_init_helper(machine_i2s_obj_t *self, size_t n_pos_args,
     init->Standard = I2S_STANDARD_PHILIPS;
     init->DataFormat = get_dma_bits(self->mode, self->bits);
     init->MCLKOutput = I2S_MCLKOUTPUT_DISABLE;
-    init->AudioFreq = args[ARG_rate].u_int;
+    init->AudioFreq = I2S_AUDIOFREQ_DEFAULT;
     init->CPOL = I2S_CPOL_LOW;
     init->ClockSource = I2S_CLOCK_PLL;
 
+    // set rate if not in passive mode
+    if (i2s_mode == I2S_MODE_MASTER_TX || i2s_mode == I2S_MODE_MASTER_RX) {
+        self->rate = args[ARG_rate].u_int;
+        if (self->rate < 0) {
+            mp_raise_ValueError(MP_ERROR_TEXT("rate required"));
+        }
+        init->AudioFreq = args[ARG_rate].u_int;
+    }
```

Comment:
To aid in code comprehension, I think this logic is better placed around line 761, in the section of code where arguments are validated.

Reply:
done